src/model/paths.ml

```diff
@@ -41,40 +47,40 @@ module Identifier = struct
     | `InstanceVariable (_, name) -> InstanceVariableName.to_string name
     | `Label (_, name) -> LabelName.to_string name
 
-  let name : [< t ] -> string = fun n -> name_aux (n :> t)
+  let name : [< t_pv ] id -> string = fun n -> name_aux (n :> t)
 
   let rec label_parent_aux =
     let open Paths_types.Identifier in
     fun (n : any) ->
       match n with
-      | `Result i -> label_parent_aux (i :> any)
-      | `CoreType _ | `CoreException _ -> assert false
-      | `Root _ as p -> (p :> label_parent)
-      | `Page _ as p -> (p :> label_parent)
-      | `LeafPage _ as p -> (p :> label_parent)
-      | `Module (p, _)
-      | `ModuleType (p, _)
-      | `Parameter (p, _)
-      | `Class (p, _)
-      | `ClassType (p, _)
-      | `Type (p, _)
-      | `Extension (p, _)
-      | `Exception (p, _)
-      | `Value (p, _) ->
+      | { iv = `Result i; _ } -> label_parent_aux (i :> any)
+      | { iv = `CoreType _; _ } | { iv = `CoreException _; _ } -> assert false
+      | { iv = `Root _; _ } as p -> (p :> label_parent)
+      | { iv = `Page _; _ } as p -> (p :> label_parent)
+      | { iv = `LeafPage _; _ } as p -> (p :> label_parent)
+      | { iv = `Module (p, _); _ }
+      | { iv = `ModuleType (p, _); _ }
+      | { iv = `Parameter (p, _); _ }
+      | { iv = `Class (p, _); _ }
+      | { iv = `ClassType (p, _); _ }
+      | { iv = `Type (p, _); _ }
+      | { iv = `Extension (p, _); _ }
+      | { iv = `Exception (p, _); _ }
+      | { iv = `Value (p, _); _ } ->
           (p : signature :> label_parent)
-      | `Label (p, _) -> p
-      | `Method (p, _) | `InstanceVariable (p, _) ->
+      | { iv = `Label (p, _); _ } -> p
+      | { iv = `Method (p, _); _ } | { iv = `InstanceVariable (p, _); _ } ->
           (p : class_signature :> label_parent)
-      | `Constructor (p, _) -> (p : datatype :> label_parent)
-      | `Field (p, _) -> (p : parent :> label_parent)
+      | { iv = `Constructor (p, _); _ } -> (p : datatype :> label_parent)
+      | { iv = `Field (p, _); _ } -> (p : parent :> label_parent)
 
   let label_parent n = label_parent_aux (n :> t)
 
-  let equal = ( = )
+  let equal x y = x.ihash = y.ihash && String.equal x.ikey y.ikey
 
-  let hash = Hashtbl.hash
+  let hash x = x.ihash
 
-  let compare = compare
+  let compare x y = compare x.ihash y.ihash
```

Comment:
This doesn't correspond to `equal`, should compare the key when the hashes matches.

Reply:
Oh, good spot thanks!

Reply:
Fixed now